Sources/_CryptoExtras/RSA/RSA_boring.swift

```diff
@@ -315,6 +344,215 @@ extension BoringSSLRSAPublicKey {
+            var blindedMessageBytes = [UInt8](repeating: 0, count: outputSize)
+            guard CCryptoBoringSSL_BN_bn2bin_padded(&blindedMessageBytes, outputSize, z) == 1 else {
+                throw CryptoKitError.internalBoringSSLError()
+            }
+
+            // 12. output blinded_msg, inv
+            let blindedMessage = Data(blindedMessageBytes)
+            var invBytes = [UInt8](repeating: 0, count: outputSize)
+            guard CCryptoBoringSSL_BN_bn2bin_padded(&invBytes, outputSize, inv) == 1 else {
+                throw CryptoKitError.internalBoringSSLError()
+            }
+            let blindInverse = _RSA.BlindSigning.BlindInverse(rawRepresentation: Data(invBytes))
+            return (blindedMessage, blindInverse)
+        }
+
+        fileprivate func finalize<H: HashFunction>(
+            _ signature: _RSA.BlindSigning.BlindSignature,
+            for message: _RSA.BlindSigning.PreparedMessage,
+            blindInverse: _RSA.BlindSigning.BlindInverse,
+            parameters: _RSA.BlindSigning.Parameters<H>
+        ) throws -> _RSA.Signing.RSASignature {
+            /// ```
+            /// All BN_CTX_get() calls must be made before calling any other functions that use the ctx as an argument.
+            /// ...
+            /// BN_CTX_get() returns a pointer to the BIGNUM, or NULL on error. Once BN_CTX_get() has failed, the
+            /// subsequent calls will return NULL as well, so it is sufficient to check the return value of the last
+            /// BN_CTX_get() call.
+            /// ```
+            /// —— Extract from `man 3 BN_CTX_get`.
+            let bnCtx = CCryptoBoringSSL_BN_CTX_new()
+            CCryptoBoringSSL_BN_CTX_start(bnCtx)
+            defer {
+                CCryptoBoringSSL_BN_CTX_end(bnCtx)
+                CCryptoBoringSSL_BN_CTX_free(bnCtx)
+            }
+            let z = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            let inv = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            let s = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            guard s != nil else { throw CryptoKitError.internalBoringSSLError() }
+
+            let rsaPublicKey = CCryptoBoringSSL_EVP_PKEY_get0_RSA(self.pointer)
+            let outputSize = Int(CCryptoBoringSSL_RSA_size(rsaPublicKey))
+
+            // 1. If len(blind_sig) != modulus_len, raise an "unexpected input size" error and stop
+            guard signature.rawRepresentation.count == outputSize else {
+                throw _RSA.BlindSigning.Error.unexpectedInputSize
+            }
+
+            // 2. z = bytes_to_int(blind_sig)
+            try signature.rawRepresentation.withUnsafeBytes { blindSignatureBytesPtr in
+                guard CCryptoBoringSSLShims_BN_bin2bn(blindSignatureBytesPtr.baseAddress, blindSignatureBytesPtr.count, z) == z else {
+                    throw CryptoKitError.internalBoringSSLError()
+                }
+            }
+
+            // 3. s = (z * inv) mod n
```

Comment:
Same note here: this is a long method, it would be nice to split it into multiple smaller methods.